models/models.reviews.js

```diff
@@ -26,15 +26,45 @@ exports.fetchReviewById = (review_id) => {
     });
 };
 
-exports.fetchReviews = () => {
-  return db
-    .query(
-      `SELECT reviews.*, count(comments.review_id)::INT AS comment_count FROM reviews
-         LEFT JOIN comments ON reviews.review_id=comments.review_id
-         GROUP BY reviews.review_id
-         ORDER BY reviews.created_at DESC;`
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+exports.fetchReviews = (sort_by = 'created_at', order = 'desc', category, reqQuery) => {
+  if (Object.keys(reqQuery).length > 0) {
+    const permittedQueries = ['sort_by', 'order', 'category'];
+
+    for (let key in reqQuery) {
+      if (!permittedQueries.includes(key)) {
+        return Promise.reject({
+          status: 400,
+          msg: 'Invalid query',
+        });
+      }
+    }
+  }
+
+  const validSort_byValues = ['review_id', 'title', 'category', 'designer', 'owner', 'review_body', 'review_img_url', 'created_at', 'votes', 'comment_count'];
+  const validSort_byorder = ['asc', 'ASC', 'desc', 'DESC'];
+  const validCategoryFilter = ['euro game', 'social deduction', 'dexterity', "children's games"];
```

Comment:
This will cause an issue when we use the development data as there are more categories than with the test data. More categories could potentially be added also. 

Instead of having a valid array like this you will have to make a query to the database to check whether the category exists, if it returns something then you're good to go and if it doesn't, promise.reject it